modulus/models/pangu/pangu.py

```diff
@@ -0,0 +1,641 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+from dataclasses import dataclass
+
+import numpy as np
+import torch
+from timm.layers import DropPath, trunc_normal_
+from torch import nn
+
+from ..meta import ModelMetaData
+from ..module import Module
+from .patch_embed import PatchEmbed2D, PatchEmbed3D, PatchRecovery2D, PatchRecovery3D
+from .shift_window_mask import get_shift_window_mask, window_partition, window_reverse
+from .utils import crop3d, get_earth_position_index, get_pad3d
+
+
+class UpSample(nn.Module):
```

Comment:
Is this class duplicated from Fengwu code? Let's consolidate all of the layers as much as possible.

Reply:
Moved to models/layers